src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDocumentMappingService.cs

```diff
@@ -491,13 +493,13 @@ public async Task<(Uri MappedDocumentUri, Range MappedRange)> MapToHostDocumentU
 
     // Internal for testing
     internal static RazorLanguageKind GetLanguageKindCore(
-        IReadOnlyList<ClassifiedSpanInternal> classifiedSpans,
-        IReadOnlyList<TagHelperSpanInternal> tagHelperSpans,
+        ImmutableArray<ClassifiedSpanInternal> classifiedSpans,
+        ImmutableArray<TagHelperSpanInternal> tagHelperSpans,
         int hostDocumentIndex,
         int hostDocumentLength,
         bool rightAssociative)
     {
-        for (var i = 0; i < classifiedSpans.Count; i++)
+        for (var i = 0; i < classifiedSpans.Length; i++)
```

Comment:
Nit: consider extracting this to a local for reuse here and below.

Reply:
Thanks!